plenoptic/simulate/canonical_computations/steerable_pyramid_freq.py

```diff
@@ -321,89 +316,98 @@ def forward(self, x, scales=[]):
 
         # x is a torch tensor batch of images of size [N,C,W,H]
         assert len(x.shape) == 4, "Input must be batch of images of shape BxCxHxW"
-        # x = x.squeeze(1) #flatten channel dimension first
-        imdft = torch.rfft(x, signal_ndim=2, onesided=False, normalized=self.fft_normalize)
-        imdft = batch_fftshift(imdft)
-
+        # if input has multiple channels, reshape the channels into the batch for independent processing
+        if x.shape[1] > 1:
+            x = x.view(x.shape[0]*x.shape[1], 1, x.shape[2], x.shape[3])
+        
+        imdft = fft.fft2(x, dim=(-2,-1), norm = self.fft_norm)
+        imdft = fft.fftshift(imdft)
         if 'residual_highpass' in scales:
             # high-pass
             hi0dft = imdft * hi0mask
-            hi0 = batch_ifftshift(hi0dft)
-            hi0 = torch.ifft(hi0, signal_ndim=2, normalized=self.fft_normalize)
-            hi0_real = torch.unbind(hi0, -1)[0]
-            self.pyr_coeffs['residual_highpass'] = hi0_real
-            self.pyr_size['residual_highpass'] = tuple(hi0_real.shape[-2:])
+            hi0 = fft.ifftshift(hi0dft)
+            hi0 = fft.ifft2(hi0, dim=(-2,-1), norm=self.fft_norm)
+            pyr_coeffs['residual_highpass'] = hi0.real
+            self.pyr_size['residual_highpass'] = tuple(hi0.real.shape[-2:])
 
+        #input to the next scale is the low-pass filtered component
         lodft = imdft * lo0mask
 
-
         for i in range(self.num_scales):
 
             if i in scales:
-
+                #high-pass mask is selected based on the current scale
                 himask = self._himasks[i]
+                #compute filter output at each orientation
                 for b in range(self.num_orientations):
+                    '''
+                    band pass filtering is done in the fourier space as multiplying by the fft of a gaussian derivative.
+                    The oriented dft is computed as a product of the fft of the low-passed component,
+                    the precomputed anglemask (specifies orientation), and the precomputed hipass mask (creating a bandpass filter)
+                    the complex_const variable comes from the Fourier transform of a gaussian derivative.
+                    Based on the order of the gaussian, this constant changes.
+                    
+                    '''
```

Comment:
minor point: we can use the `'''` as multi-line comment in the middle of a function, but I think I would prefer the `#` for consistency. Most syntax highlighters (at least, my emacs one and the github one here) seem to make the `'''` comment look like regular strings, which is confusing